### PR TITLE
Revamps the Alamo with doublewide seats and enchanced player visibility

### PR DESCRIPTION
**Pull request by ClosetedSkeletonGH**
### STOP MAKING JOKES ABOUT HAND HOLDING, I DIDNT ADD HAND HOLDING, LOOK AT THE SPRITES THERES NO HAND HOLDING. I SPECIFICALLY EDITED AND PIXEL SHIFTED THEM SO THERE WOULD BE NO MORE HAND HOLDING. IF YOU ALL KEEP SAYING HAND HOLDING i WILL GET THE MODERATORS INVOLVED!!!!!

Things added:
-bunch of new floor tiles in the old style to allow for better alamo customization

-doublewide chairs (sprites are a combo of our chairs and CM chairs, but mostly theirs), tested and functional in every aspect

-leader chairs (chairs that require a certain level of leadership to sit in)

-alamo vendors (just a snack and cola vendor that are unwrenchable and dont spam chat, normal vendors are already wrenchable so not a balance concern)

and finally, the focus of this PR:

-A brand new alamo interior, with a focus on efficiency and roleplay opportunities (IE, the center aisle wall is now all fake window + all seats are now within earshot/visual range of eachother + special leadership zone + arm touching RP + storage area is no longer hellish to move past/store things in)

pics:

From StrongDMM:
image

Alamo while empty: 
image

What the upperleft most marine can see:
image

What the bottom right most marine can see:
image

Proof the new windows wont cause major balance issues :
image

Filled almamo:
image

Oh you can also see out the front of the alamo now:
image

sprite sheet:
image

Stats, info, balance concerns:

Old Alamo:
-35 seats + 2 pilot seats
-1 attachment point
-1 holo pad
-4 intercomms
-4 fire extinguishers
-4 nanomeds
-bad, shuffle prone storage area

New Alamo:
-42 seats + 2 pilot seats + 4 SL seats + 1 FC seat
-see through aisle window (not able to be fired through by lasers though (on purpose))
-1 attatchment point
-1 holo pad (I will kill these soon in a future PR >:))
-2 intercomms (readded back two)
-0 fire extinguishers (I finalized on not including these as it would need a full resprite and code messed with, too ugly)
-4 nanomeds (this time with better visibility too
-1 alamo snack machine (not wrenchable + no audio spam)
-1 alamo soder machine (not wrenchable + no audio spam)
-good, shuffle-free storage area

misc change:
-changes the normal dropship passenger chairs bar sprite to the updated one because they are practically the same but the new one looks better. doesnt cause any sprite errors

unatomized change:
-fixes a bug with soder machine making it very hard to use, figured why not since im already touching it for the alamo

## Why It's Good For The Game

This PR attempts to redo the Alamo in a way that encourages roleplay, with every marine being on screen of eachother. With the current alamo having a huge wall blocking marines from seeing eachother, there is only 1 seat that has 80% of the marines on screen, and at max you're getting 55% of the marines on screen if youre sitting on the sides because of that wall blocking everything. 

Alamo RP is some of the silliest/best in the game and a part of the round I always enjoy. This change, along with the efficiency changes to the storage area (because we all have had a crate that gets put onto the alamo and shuffled out because having the only area for the crates in the back where people try to walk through is???), is to help encourage player interaction with eachother and player interaction with their leadership. 

## Changelog
:cl:
add: Alamo has been revamped to be more RP friendly and efficient
bugfix: fixes soder machines costing money
bugfix: fixes not being able to see out of the front of the almamo
/:cl:

### CODE REVIEW

**Comment by ivanmixo** [comment]
This is an absrudly huge buff to alamo holds, even with gas spam you'd have an extremely hard time attacking this. On the flipside it also makes it significantly harder to hold for xenomorph due to it being so open. This has much farther implications than just being a cool RP thing. Really not a fan.

**Comment by Barnet2** [comment]
Already commented to you personally but not a fan of the wall removals, double chairs are kewl though.

**Comment by DeltaFire15** [comment]
>0 intercomms

Intercomms are still (situationally) useful. They are on the frequency turrets yell on, making them somewhat of a canary on what the FOB is looking like while travelling down.
Additionally, they function while the main comms are nonfunctional (Ionospheric event, Comms outages) and are fairly neat in that regard. Besides that there's also the minor emergent aspect of being able to listen to the xenos if they control it.

> 0 fire extinguishers

Extinguisher cabinets can be nice when *someone* set the alamo on fire again, or just generally if you need one. Given, matters less to me than the intercomms.

**Comment by AwkwardStereo** [comment]
Fire extinguisher helps for when people accidentally or otherwise set it on fire. Especially during prep.

**Comment by GuyGuysonF19** [comment]
This is a bad change and shows you misunderstood the very idea of the alamo, let me demonstrate 
image

**Comment by GuyGuysonF19** [comment]
It is also worth mentioning we are not CM, and and we don't "roleplay" like that, we beat the shit out of minh on Alamo and that's it, the wall is too based to remove

**Comment by Lumipharon** [comment]
I really don't like the double seat CM thing honestly.

**Comment by FosterGrime-Heymynameslouie** [comment]
image

porrly photoshoped but I suggest something like this would be alot better
not only would you not change the old layout of alamo, you can still see and talk through the people on the other side as beccause you can still somehow hear through windows, you would also have dedicated space on the bottom for crates and mechs and such while also retaining the 1 anchor point for modules, practically eliminating all argument of "balance" concerns while also still looking very fuckin cool

**Review by Kuro020 — APPROVED** [review]
sprites lgtm

**Comment by coldironwarrior** [comment]
like this PR a lot. we do still sometimes get to the pop where there's not enough seats (between all the marines and all our Junk we take with), so the idea of better storage and more seats is really appealing.

add intercoms back. you can hear them in most CICs and i use them to spy on the xenos that get on the Alamo when i'm synth

**Comment by FosterGrime-Heymynameslouie** [comment]
really cool pr speedmerhg

**Comment by novaepee** [comment]
> added back two intercomms but when they get used for megaspam..............................

we just punch the marines who megaspam

**Comment by lbnesquik** [comment]
AI holopads don't seem to cover all the map

**Comment by Wirpy** [comment]
Yes please, finally I can sit next to amos and hold hands

**Comment by ZeWaka** [comment]
> porrly photoshoped but I suggest something like this

legalized moshpit :godmode:

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 1]
```diff
@@ -396,6 +405,119 @@
...
+/obj/structure/bed/chair/dropship/doublewide
+	name = "doublewide seat"
+	desc = "Holds you in place during high altitude drops."
+	icon_state = "doublewide_chair" //only facing south cause the rest are ugly
+	max_integrity = 130
+	var/chair_color = NO_CHAIR_COLOR
+	var/chair_state = DROPSHIP_CHAIR_UNFOLDED
+	var/leader_chair = FALSE
+	var/image/chairbar = null
+	var/buckling_x = 0 //pixel x shift to give to the buckled mob.
```
Comment (ivanmixo) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:417]: Autodoc

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 2]
```diff
@@ -396,6 +405,119 @@
...
+
+/obj/structure/bed/chair/dropship/doublewide/attack_alien(mob/living/carbon/xenomorph/X, damage_amount = X.xeno_caste.melee_damage, damage_type = BRUTE, damage_flag = "", effects = TRUE, armor_penetration = 0, isrightclick = FALSE)
+	if(X.status_flags & INCORPOREAL)
+		return FALSE
+	if(LAZYLEN(buckled_mobs))
+		unbuckle_mob(buckled_mobs[1])
+	return ..()
+
+/obj/structure/bed/chair/dropship/doublewide/attackby(obj/item/I, mob/user, params)
+	. = ..()
```
Comment (ivanmixo) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:428]: ?

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 3]
```diff
@@ -396,6 +405,119 @@
...
+	var/leader_chair = FALSE
+	var/image/chairbar = null
+	var/buckling_x = 0 //pixel x shift to give to the buckled mob.
+	buildstacktype = 0
+
+/obj/structure/bed/chair/dropship/doublewide/attack_alien(mob/living/carbon/xenomorph/X, damage_amount = X.xeno_caste.melee_damage, damage_type = BRUTE, damage_flag = "", effects = TRUE, armor_penetration = 0, isrightclick = FALSE)
+	if(X.status_flags & INCORPOREAL)
+		return FALSE
+	if(LAZYLEN(buckled_mobs))
+		unbuckle_mob(buckled_mobs[1])
```
Comment (ivanmixo) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:424]: Why not both?

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 4]
```diff
@@ -396,6 +405,119 @@
...
+	buckled_mob.old_x = initial(buckled_mob.pixel_x)
+	buckled_mob.density = TRUE
+	update_icon()
+	return ..()
+
+/obj/structure/bed/chair/dropship/doublewide/buckle_mob(mob/living/buckling_mob, force = FALSE, check_loc = TRUE, lying_buckle = FALSE, hands_needed = 0, target_hands_needed = 0, silent)
+	if(chair_state == DROPSHIP_CHAIR_BROKEN)
+		balloon_alert_to_viewers("This chair is too damaged to sit in")
+		return FALSE
+	if(leader_chair == TRUE && buckling_mob.skills.getRating(SKILL_LEADERSHIP) < SKILL_LEAD_TRAINED)
```
Comment (ivanmixo) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:472]:
```suggestion
	if(leader_chair && buckling_mob.skills.getRating(SKILL_LEADERSHIP) < SKILL_LEAD_TRAINED)
```
funny meme by the way

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 5]
```diff
@@ -396,6 +405,119 @@
...
+/obj/structure/bed/chair/dropship/doublewide/Initialize(mapload)
+	. = ..()
+	chairbar = image("icons/obj/objects.dmi", "shuttle_bars")
+	chairbar.layer = ABOVE_MOB_LAYER
+
+/obj/structure/bed/chair/dropship/doublewide/post_buckle_mob(mob/buckling_mob)
+	icon_state = "doublewide_chair[chair_color]_buckled"
+	buckling_mob.pixel_x = buckling_x
+	buckling_mob.old_x = buckling_x
+	overlays += chairbar
```
Comment (ivanmixo) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:454]: Use `update_overlays` for everything regarding updating overlays

File: code/game/turfs/open.dm [hunk 6]
```diff
@@ -146,6 +146,84 @@
 /turf/open/shuttle/dropship/fourteen
 	icon_state = "floor6"
 
+/turf/open/shuttle/dropship/fourteen
```
Comment (ivanmixo) [inline code/game/turfs/open.dm:149]: What the hell are these turfs and why?

**Review by ivanmixo — CHANGES_REQUESTED** [review]

**Comment by ivanmixo** [comment]
Also the chair type is extreme copypasta from `/obj/structure/bed/chair/dropship/passenger` why did you not just make it a subtype?

**Comment by GuyGuysonF19** [comment]
The issue is mainly the wall, people do not want the wall removed, it is more or less a useless PR. As it doesn't effectively do "anything", besides a change for the sake of change, and while sometimes that's a good thing, removing the iconic alamo wall and replacing with a box, removes character from the alamo as well as making it harder for xenos to fight in it, as their would be space for every marine on the alamo to fire on them, which is absolutely unnecessary, good faith PR, but it is just a bad PR

**Comment by Wisemonster** [comment]
> The issue is mainly the wall, people do not want the wall removed, it is more or less a useless PR. As it doesn't effectively do "anything", besides a change for the sake of change, and while sometimes that's a good thing, removing the iconic alamo wall and replacing with a box, removes character from the alamo as well as making it harder for xenos to fight in it, as their would be space for every marine on the alamo to fire on them, which is absolutely unnecessary, good faith PR, but it is just a bad PR

Except the wall is mostly there, just see-through. All the pr does is move the chairs around so they are in view of each other, and replace them with a subtype.

**Comment by breadhunt** [comment]
This is really cool. Probably melodramatic take but, I think this is would be great for RP and teamwork. 

As said previously, we're not CM, we just straight up are never going to have briefings or whatever pre-drop because it's frankly an awful rule -- what we _do_ have is the Alamo, and I think this is actually a really good change exactly for that reason. RP or otherwise, it's important to be able to talk to the guys you're about to spend an hour+ in combat with before shit hits the fan. That's not even an RP thing, this is just team stuff. Being unable to talk to everyone on the Alamo completely divides the team, which is a big issue when the Alamo is the hub for getting to know the team & what everyone's doing. It's worth remembering that little changes like this can go a long way in guiding players to work together (see DRG for a masterclass on guiding players into working as a team)

Side note, I think it's pretty cool to have the FC/etc have designated seats so that the marines can actually see who's gonna be coordinating them, rather than working off radio comms and guesswork. (Quoth the MSGT two hours into the op: "Wait, we had an FC the entire time?") Also, do you know how fucking awful it is to try to brief on Alamo currently with those walls in the way? It's hell. It's actual hell. Please let this through just for that alone oh my _god_.

**Comment by Yttriums** [comment]
built in vending machine = unlimited toki bars on alamo = marine powercreep

**Comment by chizzy376** [comment]
> image
> 
> porrly photoshoped but I suggest something like this would be alot better not only would you not change the old layout of alamo, you can still see and talk through the people on the other side as beccause you can still somehow hear through windows, you would also have dedicated space on the bottom for crates and mechs and such while also retaining the 1 anchor point for modules, practically eliminating all argument of "balance" concerns while also still looking very fuckin cool

Readd chairs to the mosh pit and give it a DJ stand

**Comment by EaglePhntm** [comment]
watch em refuse this pr but accept any removal pr till tgmc becomes pong 2.0

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 7]
```diff
@@ -260,9 +270,32 @@
 	icon_state = "officechair_dark"
 	anchored = FALSE
 
+/obj/structure/bed/chair/dropship
+	name = "dropship chair"
+	desc = "Holds you in place during high altitude drops."
+	icon_state = "shuttle_chair"
+	var/image/chairbar = null
```
Comment (ivanmixo) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:277]: doc your vars

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 8]
```diff
@@ -396,6 +406,130 @@
 		span_warning("You repair \the [src]."))
 		chair_state = DROPSHIP_CHAIR_FOLDED
 
+/obj/structure/bed/chair/dropship/doublewide
+	name = "doublewide seat"
+	icon_state = "doublewide_chair" //only facing south cause the rest are ugly
+	max_integrity = 130
+	var/chair_color = NO_CHAIR_COLOR
```
Comment (ivanmixo) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:413]: doc your vars

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 9]
```diff
@@ -396,6 +406,130 @@
...
+	doublewide_mob_density(buckling_mob, TRUE)
+	return ..()
+
+/obj/structure/bed/chair/dropship/doublewide/post_unbuckle_mob(mob/buckled_mob)
+	buckled_mob.pixel_x = initial(buckled_mob.pixel_x)
+	buckled_mob.old_x = initial(buckled_mob.pixel_x)
+	doublewide_mob_density(buckled_mob, FALSE)
+	return ..()
+
+/obj/structure/bed/chair/dropship/doublewide/proc/doublewide_mob_density(mob/interactionmob, buckling)
```
Comment (ivanmixo) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:465]: doc your procs. Also why does this need to exist? Why not just handle this in each (un)buckle proc seperately?

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 10]
```diff
@@ -396,6 +406,130 @@
...
+			//if theyre unbuckling with and there is another buckled mob
+			other_chair_buckled_mob.density = FALSE
+			interactionmob.density = TRUE
+			return
+		if(buckling) //if theyre buckling with no other chairs
+			interactionmob.density = FALSE
+			return
+	//if there is no other doublewide or theyre unbuckling
+	interactionmob.density = TRUE
+	return
```
Comment (ivanmixo) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:484]: pointless return at the end of proc

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 11]
```diff
@@ -396,6 +406,130 @@
...
+	return ..()
+
+/obj/structure/bed/chair/dropship/doublewide/post_unbuckle_mob(mob/buckled_mob)
+	buckled_mob.pixel_x = initial(buckled_mob.pixel_x)
+	buckled_mob.old_x = initial(buckled_mob.pixel_x)
+	doublewide_mob_density(buckled_mob, FALSE)
+	return ..()
+
+/obj/structure/bed/chair/dropship/doublewide/proc/doublewide_mob_density(mob/interactionmob, buckling)
+	var/obj/structure/bed/chair/dropship/doublewide/other_doublewide = locate(/obj/structure/bed/chair/dropship/doublewide) in interactionmob.loc
```
Comment (ivanmixo) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:466]: how do you know this won't just get `src`?

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 12]
```diff
@@ -396,6 +406,130 @@
...
+	var/chair_color = NO_CHAIR_COLOR
+	var/leader_chair = FALSE
+	/// pixel x shift to give to the buckled mob
+	var/buckling_x = 0
+
+/obj/structure/bed/chair/dropship/doublewide/attack_alien(mob/living/carbon/xenomorph/X, damage_amount = X.xeno_caste.melee_damage, damage_type = BRUTE, damage_flag = "", effects = TRUE, armor_penetration = 0, isrightclick = FALSE)
+	if(X.status_flags & INCORPOREAL)
+		return FALSE
+	if(LAZYLEN(buckled_mobs))
+		unbuckle_mob(buckled_mobs[1])
```
Comment (ivanmixo) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:422]:
```suggestion
		unbuckle_all_mobs(TRUE)
```

**Review by ivanmixo — CHANGES_REQUESTED** [review]

**Comment by TiviPlus** [comment]
Are we waiting for grayson on this?

**Comment by ivanmixo** [comment]
Waiting for @ClosetedSkeletonGH to address the reviews guh

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 13]
```diff
@@ -396,6 +408,132 @@
...
+
+/obj/structure/bed/chair/dropship/doublewide/welder_act(mob/living/user, obj/item/I)
+	welder_repair_act(user, I, 130, 1 SECONDS, 0, SKILL_ENGINEER_METAL, 1)
+	chair_state = DROPSHIP_CHAIR_FOLDED
+	update_icon()
+	return
+
+/obj/structure/bed/chair/dropship/doublewide/update_icon_state()
+	. = ..()
+	var/percentage = (obj_integrity / max_integrity) * 100
```
Comment (ivanmixo) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:437]: Why is this in update_icon_state

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 14]
```diff
@@ -396,6 +408,132 @@
...
+	buckled_mob.old_x = initial(buckled_mob.pixel_x)
+	doublewide_mob_density(buckled_mob, FALSE)
+	return ..()
+
+/// Handles checking the changing density for the person buckling or unbuckling and the person next to the buckling/unbuckling person
+/obj/structure/bed/chair/dropship/doublewide/proc/doublewide_mob_density(mob/interactionmob, buckling)
+	var/obj/structure/bed/chair/dropship/doublewide/other_doublewide = locate(/obj/structure/bed/chair/dropship/doublewide) in interactionmob.loc
+	var/mob/living/other_chair_buckled_mob
+	if(other_doublewide) //if there is another doublewide
+		if((LAZYLEN(other_doublewide.buckled_mobs)) && (other_doublewide.buckled_mobs[1] != interactionmob))
```
Comment (ivanmixo) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:474]:
```suggestion
		if(LAZYLEN(other_doublewide.buckled_mobs) && (other_doublewide.buckled_mobs[1] != interactionmob))
```

**Review by ivanmixo — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm (hunk 13 above)
Comment (GrrrKitten) [reply code/game/objects/structures/stool_bed_chair_nest/chairs.dm:437]: icon state is handled by health, not sure where else to put this, don't really remember the precedent

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 15]
```diff
@@ -396,6 +408,135 @@
...
+
+/obj/structure/bed/chair/dropship/doublewide/update_icon_state()
+	. = ..()
+	var/percentage = (obj_integrity / max_integrity) * 100
+	switch(percentage)
+		if(-INFINITY to 65)
+			chair_state = DROPSHIP_CHAIR_BROKEN
+	icon_state = "doublewide_chair[chair_color][chair_state]"
+
+/obj/structure/bed/chair/dropship/doublewide/attackby(obj/item/I, mob/user, params)
```
Comment (ivanmixo) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:443]: You sure you don't need an update_icon call as well?

Comment (ivanmixo) [reply code/game/objects/structures/stool_bed_chair_nest/chairs.dm:443]: If you say so

**Review by ivanmixo — APPROVED** [review]

**Review by Neocloudy — COMMENTED** [review]
image
spellcheck :D on all chair examine strings

**Comment by TiviPlus** [comment]
Is the SL chairs being destructible intended

**Comment by ivanmixo** [comment]
why this not merged yet

**Comment by EaglePhntm** [comment]
> why this not merged yet

tgmc speedmerging every terrible pr, meanwhile good prs....

**Comment by coldironwarrior** [comment]
> why this not merged yet

image

**Comment by Lumipharon** [comment]
Old chair:1 slash
New chair: 12 slashes or so per chair, with 2 per tile.

**Pull request merged**